devenv/bootstrap.py

```diff
@@ -12,15 +12,24 @@
 from devenv.lib import direnv
 from devenv.lib import github
 from devenv.lib import proc
+from devenv.lib.config import Config
+from devenv.lib.config import DEFAULT_CONFIG
+from devenv.lib.config import initialize_config
 
 help = "Bootstraps the development environment."
 ExitCode: TypeAlias = "str | int | None"
 
 
-def main(coderoot: str, argv: Sequence[str] | None = None) -> ExitCode:
+def main(
+    config_path: str, coderoot: str, argv: Sequence[str] | None = None
+) -> ExitCode:
     parser = argparse.ArgumentParser(description=help)
     parser.parse_args(argv)
 
+    default_config: Config = {**DEFAULT_CONFIG}
+    default_config["devenv"].update({"coderoot": coderoot})
```

Comment:
ah while looking at the test this reminds me of another problem here -- `{**d}` is a shallow copy:

```pycon
>>> DEFAULT_CONFIG = {"whatever": 1, "devenv": {"coderoot": "wat"}}
>>> other = {**DEFAULT_CONFIG}
>>> other["devenv"]["coderoot"] = "something-else"
>>> DEFAULT_CONFIG  # oops!
{'whatever': 1, 'devenv': {'coderoot': 'something-else'}}
```

probably needs a deepcopy -- or `DEFAULT_CONFIG` should maybe be a function which *returns* a fresh dict each time?

Reply:
This line is gone, however, thanks for the reminder to watch my nested structures. I think DEFAULT_CONFIG can be deleted now -- I'll do that as the action here.

Reply:
Constant removed fully.